parlai/core/torch_agent.py

```diff
@@ -1708,19 +1710,26 @@ def batch_act(self, observations):
         ``eval_step`` methods instead. The former is called when labels are
         present in the observations batch; otherwise, the latter is called.
         """
-        batch_size = len(observations)
+        if not isinstance(observations, Batch):
+            batch_size = len(observations)
+        else:
+            batch_size = observations.text_vec.size(0)
```

Comment:
This seems like a good opportunity to add that batchsize field you’ve always wanted